students/thorn/lesson03/mailroom.py

```diff
@@ -26,33 +26,69 @@ def send_thanks():
     - Print thank you note
     - Returns to original prompt or quit to prompt
     """
-    pass
+    name_request = input("Please enter a donor's full name, or 'list' for a full list of donors.  'Quit' or 'Return' valid commands. ")
 
+    # Return or Quit.
+    if name_request.lower() == "quit":
+        quit()
+
+    if name_request.lower() == "return":
+        return
+
+    # List request - only donors returned.
+    if name_request.lower() == 'list':
+        for donors in DONOR_HISTORY:
+            print(donors[0])
+        return
+
+    name_request = name_request.title()
+    donation_amount = float(input(("Please enter a donation amount for {}: ".format(name_request.title()))))
+
+    # Add to existing donor.  i is the donor location in the nested list.
+    for i, donors in enumerate(DONOR_HISTORY):
+        if donors[0].title() == name_request.title():
+            DONOR_HISTORY[i][1].append(donation_amount)
+            # Break to stop it from creating i numbers of the person.
+            break
+        # Create new donor.
+        else:
+            DONOR_HISTORY.append([name_request.title(), [donation_amount]])
+            break
```

Comment:
you don't need enumerate, here is a simplified version:
```
for donor in donors:
    if donor[0].title() == name_request:  # name_request already has title applied
        donor[1].append(donation_amount)
```

your `else` has a bug. run your program, select send thanks then enter David Beckham and donation amount, now select create report - notice that david beckham is listed twice. Can you spot why?

Reply:
Thanks for all the suggestions!  I changed this to add the donation amount to an existing donor if its found (in your suggested syntax), print the message and return.  Then outside of that it takes care of adding a new donor if an existing one was not found.  

It seems to be working correctly - but was this the solution you had in mind?  Thanks again!

    for donor in donors:
        if donor[0].title() == name_request:
            donor[1].append(donation_amount)
            print(f"Thank you {name_request.title()} for your donation of ${donation_amount:.2f}.")
            return

    donors.append([name_request, [donation_amount]])